### ISSUE TITLE
question regarding similar libraries

### ISSUE THREAD

**Issue by hannesdelbeke** [tahv/pyforce#3]
hi

thank you for releasing your perforce python module.
have you encountered python-perforce, and if yes, what are the differences with your library?
- https://pypi.org/project/python-perforce/
- https://github.com/theiviaxx/python-perforce

**Comment by tahv** [comment]
Yes, python-perforce is one of the inspiration for this package. 
Both are subprocessing `p4` and use the `marshal` module.

The main differences that come to mind are:

- pyforce is fully typed, making it more convenient to use with a Language Server Protocol (LSP).
- pyforce uses [pydantic](https://docs.pydantic.dev/latest/) for data deserialization and validation.
- python-perforce supports Python 2, pyforce does not.

Additionally, the exposed commands and models might differ. Given the vastness of Perforce, pyforce does not cover every possible command. I included the commands and models I use most frequently. However, both pyforce and python-perforce allow you to execute custom commands via `pyforce.p4` and `perforce.Connection.run`, respectively.

In addition, pyforce test suite run in the CI pipeline and can be executed locally with the provided Dockerfile.

I hope this helps.
